### PR TITLE
ci: Add twister tag support to get_twister_opt

### PR DESCRIPTION
**Pull request by galak**
Add the ability to map file/dir paths of a PR to twister TAG.  We
introduce scripts/ci/tags.yaml to conveys which files are associated
with which tag.

Since not all file/tags will be specified in tags.yaml we use the
combination of the files modified list and the tags.yaml information to
determine which tags can be excluded (ie if the file list doesn't match
any file in tags.yaml for a given tag listed there, we can that exclude
it).

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by galak** [comment]
This still needs to be cleaned up and plumbed in to the ci scripts.

The main idea here is around trying to reduce how often we build kernel tests.  If a PR touches a driver, building all the kernel tests is kinda pointless.

File: scripts/ci/tags.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+kernel:
+    files:
```
Comment (galak) [inline scripts/ci/tags.yaml:2]: Need to define what the file set here would actually be.

**Comment by galak** [comment]
Based on PR #36065

File: scripts/ci/tags.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#
+# All tags must have a 'files' and/or 'files-regex' key.
+
+kernel:
+    files:
+        - kernel/
+        - include/*.h
+        - include/app_memory/
+        - include/arch/
+        - include/ipc/
```
Comment (nashif) [inline scripts/ci/tags.yaml:45]: this one is not related to kernel

Comment (galak) [reply scripts/ci/tags.yaml]: fixed.

**Review by nashif — APPROVED** [review]

**Review by PerMac — APPROVED** [review]
The logic in this PR looks good to me. However, I see some pitfalls we have to be aware of and worth monitoring:
The CI will skip any test which has 'kernel' tag in testcase/sample.yaml for PRs that do not change any file from the list from tag.yaml under 'kernel'. I see a potential issue if a test has more than a single tag in its yaml. It might cause over-eagerly disabling such tests. I don't think it is a major concern, just something to be aware of and adding a proper patch if needed (e.g. if a test has more tags, all of them have to be matched to get the test disabled).
